dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/AnalyticalObject.java

```diff
@@ -83,4 +85,9 @@ public interface AnalyticalObject
   boolean hasUserOrgUnit();
 
   void clearTransientState();
+
+  @Nullable
+  default LegendDefinitions getLegendDefinitions() {
+    return null;
+  }
```

Comment:
This default method is added in order to fix the issue but not breaking legacy object model